tests/unit/fem/test_pgridfunc_save_serial.cpp

```diff
@@ -14,79 +14,66 @@
 
 using namespace mfem;
 
-namespace pgridfunc_save_in_serial
-{
-
-double squared(const Vector &x)
-{
-   double sum = 0.0;
-   for (int d = 0; d < x.Size(); d++)
-   {
-      sum += std::pow(x(d), 2.0);
-   }
-   return sum;
-}
-
-
 #ifdef MFEM_USE_MPI
-#
-TEST_CASE("ParGridFunction in Serial",
-          "[ParGridFunction]"
-          "[Parallel]")
-{
-   int num_procs;
-   MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
 
-   int my_rank;
-   MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
+TEST_CASE("ParGridFunction in Serial", "[ParGridFunction][Parallel]")
+{
+   const int num_procs = Mpi::WorldSize();
+   const int my_rank = Mpi::WorldRank();
 
-   int order = 1;
-   int save_rank = 0;
-   int n = 2 * num_procs;
+   const int order = 1;
+   const int save_rank = 0;
 
-   FunctionCoefficient squaredFC(squared);
+   const int n = 2 * num_procs;
    Mesh mesh = Mesh::MakeCartesian2D(n, n, Element::QUADRILATERAL);
    mesh.SetCurvature(2);
-   H1_FECollection fec(order, mesh.Dimension());
-
-   double ser_l2_err = 0.0;
-   FiniteElementSpace fespace(&mesh, &fec);
-   GridFunction x(&fespace);
-   x.ProjectCoefficient(squaredFC);
-   ser_l2_err = x.ComputeL2Error(squaredFC);
 
    // Define a parallel mesh by a partitioning of the serial mesh.
    ParMesh pmesh(MPI_COMM_WORLD, mesh);
+   H1_FECollection fec(order, mesh.Dimension());
    ParFiniteElementSpace pfespace(&pmesh, &fec);
    ParGridFunction px(&pfespace);
-   px.ProjectCoefficient(squaredFC);
+   px.Randomize(1);
+   // Ensure that the L-DOFs are set consistently on all ranks
+   px.SetTrueVector();
+   px.SetFromTrueVector();
+
+   ConstantCoefficient zero(0.0);
+   const double l2_norm = px.ComputeL2Error(zero);
 
    // Get the ParMesh and ParGridFunction on 1 of the mpi ranks. Check the
    // L2 error on that rank and save gridfunction.
    Mesh par_to_ser_mesh = pmesh.GetSerialMesh(save_rank);
-   GridFunction x_par_to_ser = px.GetSerialGridFunction(save_rank,
-                                                        par_to_ser_mesh);
-   double par_to_ser_l2_err = 0.0;
+   GridFunction x_par_to_ser = px.GetSerialGridFunction(
+                                  save_rank, par_to_ser_mesh);
```

Comment:
I am aware we are already testing the `GetSerialGridFunction(int save_rank, FiniteElementSpace &serial_fes)` method implicitly here but it might be worth explicitly constructing another `GridFunction` using it and test its norm to demonstrate usage.

Reply:
Done in afbd1a1ba590ade194e42b86f7873daaa23e6547.